dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -1360,10 +1373,18 @@ private String getFromSubQueryLimitAndOffset( TrackedEntityInstanceQueryParams p
     {
         StringBuilder limitOffset = new StringBuilder();
         int limit = params.getMaxTeiLimit();
+        int teiQueryLimit = systemSettingManager.getIntSetting( SettingKey.TRACKED_ENTITY_MAX_LIMIT );
 
         if ( limit == 0 && !params.isPaging() )
         {
-            return "";
+
+            return Optional.of( teiQueryLimit ).filter( a -> teiQueryLimit > 0 ).map( a -> limitOffset
```

Comment:
This one https://github.com/dhis2/dhis2-core/pull/11321/files/3f2d605b48a848b713fe96f2d85c88e2cb5049bc#r955991464
is not resolved.

Pasting it here in case it got lost for you 😄 

I think we should simplify this into

```java
        if (teiQueryLimit > 0) {
            limitOffset
                    .append("limit")
                    .append(" ")
                    .append(teiQueryLimit)
                    .append(" ");
        }

        return limitOffset.toString();
```

I also assume a limit `teiQueryLimit < 0` does not make sense hence the `teiQueryLimit > 0` check.

Reply:
existing code is doing the same thing with the only difference of using `Optional` instead of `If` statement

Reply:
I really think that the if version is much more readable.
@zubaira You are right that they are doing the same thing but the use of Optional here is just confusing.
Optional should be used for variables that maybe are present and maybe they are not (null), in this case `teiQueryLimit ` can never be null, especially because it is an `int`.
What you want here is to do something if `teiQueryLimit` is greater than 0 and do nothing otheerwise.
An if statement here is much more human readable than the use of Optional